src/cli.rs

```diff
@@ -0,0 +1,733 @@
+    pub base_directory: Option<PathBuf>,
+    /// the search pattern (a regular expression, unless '--glob' is used; optional)
+    ///
+    /// the search pattern which is either a regular expression (default) or a glob
+    /// pattern (if --glob is used). If no pattern has been specified, every entry
+    /// is considered a match. If your pattern starts with a dash (-), make sure to
+    /// pass '--' first, or it will be considered as a flag (fd -- '-foo').
+    #[clap(value_parser, default_value = "", hide_default_value = true)]
+    pub pattern: String,
+    /// Set path separator when printing file paths
+    /// Set the path separator to use when printing file paths. The default is
+    /// the OS-specific separator ('/' on Unix, '\\' on Windows).
+    #[clap(long, value_name = "separator", hide_short_help = true, action)]
+    pub path_separator: Option<String>,
+    /// the root directories for the filesystem search (optional)
+    ///
+    /// The directories where the filesystem search is rooted.
+    /// If omitted, search the current working directory.
+    #[clap(action = ArgAction::Append)]
+    path: Vec<PathBuf>,
+    /// Provides paths to search as an alternative to the positional <path> argument
+    ///
+    /// Changes the usage to `fd [OPTIONS] --search-path <path> --search-path <path2> [<pattern>]`
+    #[clap(long, conflicts_with("path"), action, hide_short_help = true)]
+    search_path: Vec<PathBuf>,
+    /// strip './' prefix from non-tty outputs
+    ///
+    /// By default, relative paths are prefixed with './' when the output goes to a non
+    /// interactive terminal (TTY). Use this flag to disable this behaviour.
+    #[clap(long, conflicts_with_all(&["path", "search-path"]), hide_short_help = true, action)]
+    pub strip_cwd_prefix: bool,
+    /// Filter by owning user and/or group
+    ///
+    /// Filter files by their user and/or group. Format: [(user|uid)][:(group|gid)].
+    /// Either side is optional. Precede either side with a '!' to exclude files instead.
+    ///
+    /// Examples:
+    ///     --owner john
+    ///     --owner :students
+    ///     --owner '!john:students'
+    #[cfg(unix)]
+    #[clap(long, short = 'o', value_parser = OwnerFilter::from_string, value_name = "user:group", verbatim_doc_comment)]
+    pub owner: Option<OwnerFilter>,
+    /// Do not descend into a different file system
+    ///
+    /// By default, fd will traverse the file system tree as far as other options
+    /// dictate. With this flag, fd ensures that it does not descend into a
+    /// different file system than the one it started in. Comparable to the -mount
+    /// or -xdev filters of find(1).
+    #[cfg(any(unix, windows))]
+    #[clap(long, aliases(&["mount", "xdev"]), hide_short_help = true)]
+    pub one_file_system: bool,
+
+    #[cfg(feature = "completions")]
+    #[clap(long, value_parser = value_parser!(Shell), hide = true, exclusive = true)]
```

Comment:
nit: `#[clap(long, value_parser = value_parser!(Shell)` should be the default behavior if you opt-in to the new behavior with `#[clap(value_parser)]` or `#[clap(action)]`.

Also, `clap_complete::Shell` implements `ValueEnum`, so you could just specify `value_enum`